.github/workflows/release_ruleset.yaml

```diff
@@ -55,31 +74,76 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ inputs.tag_name }}
+          path: this
 
-      - uses: bazel-contrib/setup-bazel@0.8.0
+      - uses: bazel-contrib/setup-bazel@0.14.0
         with:
+          module-root: this
           disk-cache: ${{ inputs.mount_bazel_caches }}
           external-cache: ${{ inputs.mount_bazel_caches }}
           repository-cache: ${{ inputs.mount_bazel_caches }}
 
       - name: Test
+        working-directory: this
         run: ${{ inputs.bazel_test_command }} --disk_cache=~/.cache/bazel-disk-cache --repository_cache=~/.cache/bazel-repository-cache
 
-      - name: Build release artifacts and prepare release notes
+      - name: Release preparation
+        id: release_prep
+        working-directory: this
         run: |
           if [ ! -f "${{ inputs.release_prep_command }}" ]; then
             echo "ERROR: create a ${{ inputs.release_prep_command }} release prep script or configure a different release prep command with the release_prep_command attribute"
             exit 1
           fi
-          ${{ inputs.release_prep_command }} ${{ inputs.tag_name || github.ref_name }} > release_notes.txt
+
+          export ARTIFACTS_DIR=$(mktemp --directory)
+          export RELEASE_NOTES=$(mktemp)
+          OUTPUT=$(${{ inputs.release_prep_command }} ${{ inputs.tag_name || github.ref_name }} | jq --compact-output .)
+
+          echo "Release prep output:"
```

Comment:
I'm curious if we can get around this release_prep.sh mechanism. Like how much variance are we truly seeing with module builds -- ideally this action is responsible for the archiving as much as possible. I'll have to check what qualifies us for higher quality attestations.